KnitteryTests/QueryTests.swift

```diff
@@ -0,0 +1,115 @@
+        query.search = "cowl"
+        query.invert = true
+        let queryString = QueryBuilder.build(query)
+        
+        XCTAssertEqual(queryString, "?query=cowl$sort=_best")
+    }
+    
+    func testSecondPageQuery() {
+        query.clear()
+        query.search = "cowl"
+        query.page = "2"
+        let queryString = QueryBuilder.build(query)
+        
+        XCTAssertEqual(queryString, "?query=cowl$sort=best$page=2")
+    }
+    
+    func testSingleNotebookQuery() {
+        query.clear()
+        query.search = "cowl"
+        query.notebook.append(.favorites)
+        let queryString = QueryBuilder.build(query)
+        
+        XCTAssertEqual(queryString, "?query=cowl$sort=best$notebook-p=faved")
+    }
+    
+    func testMultipleNotebooksQuery() {
+        query.clear()
+        query.search = "cowl"
+        query.notebook.append(.favorites)
+        query.notebook.append(.library)
+        let queryString = QueryBuilder.build(query)
+        
+        XCTAssertEqual(queryString, "?query=cowl$sort=best$notebook-p=faved%7Clibrary")
+    }
+    
+    func testSingleCraftQuery() {
+        query.clear()
+        query.search = "hat"
+        query.craft.append(QCraft.crochet)
+        let queryString = QueryBuilder.build(query)
+        
+        XCTAssertEqual(queryString, "?query=hat$sort=best$craft=crochet")
+    }
+    
+    func testMultipleCraftsQuery() {
+        query.clear()
+        query.search = "hat"
+        query.craft.append(QCraft.knitting)
+        query.craft.append(QCraft.crochet)
+        let queryString = QueryBuilder.build(query)
+        
+        XCTAssertEqual(queryString, "?query=hat$sort=best$craft=knitting%7Ccrochet")
+    }
+    
+    func testSingleAvailabilityQuery() {
+        query.clear()
+        query.search = "cowl"
+        query.availability.append(.nocost)
+        let queryString = QueryBuilder.build(query)
+        
+        XCTAssertEqual(queryString, "?query=cowl$sort=best$availability=free")
+    }
+    
+    func testDoubleAvailabilitiesQuery() {
+        query.clear()
+        query.search = "cowl"
+        query.availability.append(.nocost)
+        query.availability.append(.ravelryDownload)
+        let queryString = QueryBuilder.build(query)
+        
+        XCTAssertEqual(queryString, "?query=cowl$sort=best$availability=free%7Cravelry")
+    }
+    
+    func testSingleWeightQuery() {
+        query.clear()
+        query.search = "cowl"
+        query.weight.append(.fingering)
+        let queryString = QueryBuilder.build(query)
+        
+        XCTAssertEqual(queryString, "?query=cowl$sort=best$weight=fingering")
+    }
+    
+    func testDoubleWeightsQuery() {
+        query.clear()
+        query.search = "cowl"
+        query.weight.append(.fingering)
+        query.weight.append(.jumbo)
+        let queryString = QueryBuilder.build(query)
+        
+        XCTAssertEqual(queryString, "?query=cowl$sort=best$weight=fingering%7Cjumbo")
+    }
```

Comment:
Looks like we are missing a test to test `sort`

Reply:
Sort requires no standalone test as it is included in every query with testBasicInvertQuery() testing its inversion property